### PR TITLE
chore(CON13-02): Remove `make_ecdsa_signing_subnet_list_key` from the codebase

### PR DESCRIPTION
**Pull request by Sawchord**
There was a test in consensus `framework` left over, that was still not fully migrated to use `make_chain_key_signing_subnet_list`.

This MR migrates the test, (even though it seems we are not even using Threshold Sigs in these tests), and can then remove `make_ecdsa_signing_subnet_list_key` and its test altogether.

### CODE REVIEW

File: rs/consensus/tests/framework/mod.rs [hunk 1]
```diff
@@ -56,19 +58,38 @@ pub fn setup_subnet<R: Rng + CryptoRng>(
     let registry_version = RegistryVersion::from(initial_version);
     let data_provider = Arc::new(ProtoRegistryDataProvider::new());
     let registry_client = Arc::new(FakeRegistryClient::new(Arc::clone(&data_provider) as Arc<_>));
-    let ecdsa_key_id = EcdsaKeyId {
-        curve: EcdsaCurve::Secp256k1,
-        name: "test_key".to_string(),
-    };
+
+    let threshold_keys = vec![
```
Comment (eichhorl) [inline rs/consensus/tests/framework/mod.rs:62]:
```suggestion
    let threshold_keys_ids = vec![
```

File: rs/consensus/tests/framework/mod.rs [hunk 2]
```diff
@@ -56,19 +58,38 @@ pub fn setup_subnet<R: Rng + CryptoRng>(
...
-            key_configs: vec![KeyConfig {
-                key_id: MasterPublicKeyId::Ecdsa(ecdsa_key_id.clone()),
-                pre_signatures_to_create_in_advance: 4,
-                max_queue_size: 40,
-            }],
+            // key_configs: vec![KeyConfig {
+            //     key_id: ecdsa_key_id.clone(),
+            //     pre_signatures_to_create_in_advance: 4,
+            //     max_queue_size: 40,
+            // }],
```
Comment (eichhorl) [inline rs/consensus/tests/framework/mod.rs:84]:
```suggestion
```

File: rs/consensus/tests/framework/mod.rs [hunk 3]
```diff
@@ -177,17 +198,18 @@ pub fn setup_subnet<R: Rng + CryptoRng>(
...
+    for key_id in threshold_keys {
+        data_provider
+            .add(
+                &ic_registry_keys::make_chain_key_signing_subnet_list_key(&key_id),
+                registry_version,
+                Some(ic_protobuf::registry::crypto::v1::EcdsaSigningSubnetList {
+                    subnets: vec![subnet_id_into_protobuf(subnet_id)],
+                }),
+            )
+            .expect("Could not add ECDSA signing subnet list");
```
Comment (eichhorl) [inline rs/consensus/tests/framework/mod.rs:211]:
```suggestion
            .expect("Could not add chain key signing subnet list");
```

File: rs/consensus/tests/framework/mod.rs [hunk 4]
```diff
@@ -56,19 +58,38 @@ pub fn setup_subnet<R: Rng + CryptoRng>(
...
 
     let subnet_record = SubnetRecordBuilder::from(node_ids)
         .with_dkg_interval_length(19)
         .with_chain_key_config(ChainKeyConfig {
-            key_configs: vec![KeyConfig {
-                key_id: MasterPublicKeyId::Ecdsa(ecdsa_key_id.clone()),
-                pre_signatures_to_create_in_advance: 4,
-                max_queue_size: 40,
-            }],
+            // key_configs: vec![KeyConfig {
```
Comment (chmllr) [inline rs/consensus/tests/framework/mod.rs:80]: Leftover?

File: rs/consensus/tests/framework/mod.rs [hunk 5]
```diff
@@ -177,17 +198,18 @@ pub fn setup_subnet<R: Rng + CryptoRng>(
...
-        )
-        .expect("Could not add ECDSA signing subnet list");
-
+    // Add threshold signing subnet to registry
+    for key_id in threshold_keys {
+        data_provider
+            .add(
+                &ic_registry_keys::make_chain_key_signing_subnet_list_key(&key_id),
+                registry_version,
+                Some(ic_protobuf::registry::crypto::v1::EcdsaSigningSubnetList {
```
Comment (eichhorl) [inline rs/consensus/tests/framework/mod.rs:207]:
```suggestion
                Some(ic_protobuf::registry::crypto::v1::ChainKeySigningSubnetList {
```

**Review by eichhorl — APPROVED** [review]

**Comment by eichhorl** [comment]
> (even though it seems we are not even using Threshold Sigs in these tests)

I think you are right that we aren't requesting any _signatures_ in integrations tests. But we do have tests for key generation, also under the presence of some malicious behavior IIRC. Maybe we can see if those need to be extended for Schnorr as well

**Review by aterga — APPROVED** [review]

File: rs/consensus/tests/framework/mod.rs (hunk 4 above)
Comment (Sawchord) [reply rs/consensus/tests/framework/mod.rs:80]: Correct

**Comment by Sawchord** [comment]
> Maybe we can see if those need to be extended for Schnorr as well

You are right, we have two tests that use the `ecdsa` key:
- `ecdsa_pubkey_is_produced`
- `minority_maliciouly_ecdsa_dealers_would_pass`

The test is not adapted for multiple keys yet. It simply checks that the list of keys is not empty. So any of the three keys will make the test pass, only no keys will fail.

In a follow up PR, I will make the tests more precise.

**Pull request merged**